src/components/forms/sendEmailForm/index.tsx

```diff
@@ -1,53 +1,112 @@
+import { useTranslation } from 'react-i18next';
+import { n } from '../../../utils/stringFormat';
+import DOMPurify from 'isomorphic-dompurify';
+
+const PreviewSwitch = styled(Switch)`
+  display: flex;
+  align-items: center;
+  margin-bottom: 10px;
+  padding: 10px 5px;
+`;
+
+const EmailPreview = styled.div`
+  min-height: 250px;
+  margin-bottom: 24px;
+  max-height: 800px;
+  border: 1px solid #d9d9d9;
+  border-radius: 4px;
+  padding: 3px 11px;
+  resize: vertical;
+  overflow-y: scroll; // required for resizing
+`;
 
 interface SendEmailFormProps {
   readonly emails: string[];
 }
 
 const SendEmailForm: React.FC<SendEmailFormProps> = ({ emails }) => {
+  const { t } = useTranslation(n(site, ['forms']), {
+    keyPrefix: 'volunteer_emailer',
+    nsMode: 'fallback',
+  });
+
   const [sendEmailForm] = Form.useForm();
 
+  const [showPreview, setShowPreview] = useState<boolean>(false);
+  const [bodyContent, setBodyContent] = useState<string>('');
+  const [sanitizedBodyContent, setSanitizedBodyContent] = useState<string>('');
+
+  const togglePreview = (isShowPreview: boolean) => {
+    setShowPreview(isShowPreview);
+    if (isShowPreview) setSanitizedBodyContent(DOMPurify.sanitize(bodyContent));
+  };
+
   const onFinishSendEmail = (values: SendEmailFormValues) => {
+    if (emails.length === 0) {
+      message.error(t('users_required'));
+      return;
+    }
+
     const sendEmailRequest: SendEmailRequest = {
       ...values,
       emails,
     };
 
     ProtectedApiClient.sendEmail(sendEmailRequest)
       .then(() => {
-        message.success('Email sent!');
+        message.success(t('success'));
       })
       .catch((err) =>
-        message.error(`Email could not be sent: ${err.response.data}`),
+        message.error(t('response_error', { error: err.response.data })),
       );
   };
 
   return (
-    <Form name="sendEmail" form={sendEmailForm} onFinish={onFinishSendEmail}>
+    <Form
+      name="sendEmail"
+      form={sendEmailForm}
+      onFinish={onFinishSendEmail}
+      onValuesChange={(changedValues, _) => {
+        if (changedValues.emailBody !== undefined)
+          setBodyContent(changedValues.emailBody);
+      }}
+    >
       <Form.Item
         name="emailSubject"
-        rules={requiredRule('The email subject is required')}
+        rules={requiredRule(t('subject_required'))}
       >
-        <Input placeholder="Email Subject" />
+        <Input placeholder={t('subject_placeholder')} />
       </Form.Item>
-
+      <PreviewSwitch
+        onChange={togglePreview}
+        checkedChildren={t('preview.preview')}
+        unCheckedChildren={t('preview.raw')}
+      />
       <Form.Item
         name="emailBody"
-        rules={requiredRule('The email body is required')}
+        rules={requiredRule(t('body_required'))}
+        hidden={showPreview}
       >
-        <Input.TextArea rows={6} placeholder="Email Body" />
+        <Input.TextArea rows={8} placeholder={t('body_placeholder')} />
```

Comment:
What's the reason to set the row to 8? if the user has less than 8 lines, would it just show a bunch of white spaces?

Reply:
Yep, it's just making the textbox for the the email body a little bigger (see pic). No real important reason behind this, I just thought it'd be nice to have a little more space to view the email HTML (before/after):

![image](https://github.com/Code-4-Community/speak-for-the-trees-frontend/assets/68914997/88bcadfb-da5b-4c79-8468-4dc3faea3345)


![image](https://github.com/Code-4-Community/speak-for-the-trees-frontend/assets/68914997/2ac857b5-364b-4853-ab2d-bed84b0f2401)